pkg/controller/endpointslice/endpointslice_controller_test.go

```diff
@@ -274,6 +274,31 @@ func TestSyncServiceEndpointSliceLabelSelection(t *testing.T) {
 	cmc.Check(t)
 }
 
+func TestOnEndpointSliceUpdate(t *testing.T) {
+	_, esController := newController([]string{"node-1"}, time.Duration(0))
+	ns := metav1.NamespaceDefault
+	serviceName := "testing-1"
+	epSlice1 := &discovery.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "matching-1",
+			Namespace: ns,
+			Labels: map[string]string{
+				discovery.LabelServiceName: serviceName,
+				discovery.LabelManagedBy:   controllerName,
+			},
+		},
+		AddressType: discovery.AddressTypeIPv4,
+	}
+
+	epSlice2 := epSlice1.DeepCopy()
+	epSlice2.Labels[discovery.LabelManagedBy] = "something else"
+
+	assert.Equal(t, 0, esController.queue.Len())
+	esController.onEndpointSliceUpdate(epSlice1, epSlice2)
+	time.Sleep(1 * time.Second)
+	assert.Equal(t, 1, esController.queue.Len())
```

Comment:
Unfortunately the test failures on this PR make it look like this delay will end up having flaky results. There may be a better more reliable way to see if something has been queued. One option might be to use `wait.PollImmediate()` to check every second for up to 3 seconds if something has been added to the queue and return true if it has.

Reply:
I change to use wait.PollImmediate() to check every 100ms for up to 3 seconds.

Reply:
This looks good, thanks! It looks like you just need to run `hack/update-bazel.sh` to get the verify test to pass now. Also once you've done that do you mind rebasing and squashing this back into a single commit for the PR? Thanks for your work on this!